scripts/batch_file_archiver/batch_file_archiver.py

```diff
@@ -0,0 +1,67 @@
+from typing import List
+import json
+import sys
+from multiprocessing.dummy import Pool
+import subprocess
+from typing import Dict
+from dataclasses import dataclass
+
+
+@dataclass
+class ArchiveJob:
+    job_dict: Dict
+    job_file_path: str
+    job_stdout_path: str
+
+
+class BatchFileArchiver:
+
+    def __init__(self, aap_username: str, aap_password: str, aws_access_key: str, aws_key_secret: str):
+        self.aap_username = aap_username
+        self.aap_password = aap_password
+        self.aws_access_key = aws_access_key
+        self.aws_key_secret = aws_key_secret
+
+    def run(self, archive_jobs: Dict):
+        split_jobs = self.split_archive_jobs(archive_jobs)
+        Pool(len(split_jobs)).map(lambda job: self.run_job(job), split_jobs)
+
+    def run_job(self, archive_job: ArchiveJob):
+        self.write_job_json(archive_job)
+        job_bsub_command = self.bsub_command(archive_job)
+        self.run_command(job_bsub_command, stdout_filename=archive_job.job_stdout_path)
+
+    @staticmethod
+    def run_command(cmd_string: str, stdout_filename: str):
+        with open(stdout_filename, "w") as stdout_file:
+            subprocess.Popen(cmd_string, shell=True, stdout=stdout_file)
+
+    @staticmethod
+    def split_archive_jobs(archive_jobs: Dict) -> List[ArchiveJob]:
+        return [ArchiveJob({"jobs": [job]}, f'job{i}.json', f'stdout_job{i}.txt') for (i, job) in enumerate(archive_jobs["jobs"])]
+
+    @staticmethod
+    def write_job_json(archive_job: ArchiveJob) -> str:
+        with open(archive_job.job_file_path, "w") as job_dict_file:
+            json.dump(archive_job.job_dict, job_dict_file)
+            return job_dict_file.name
+
+    def bsub_command(self, archive_job: ArchiveJob) -> str:
+        return f' bsub -M 64000 "{self.singularity_command(archive_job)}"'
```

Comment:
I think it would be better to remove the `-M 64000`. There's a chance that it won't be scheduled quickly. Let the user rerun the command if job runs out of memory. The command will be emailed.

Reply:
But if they re-run won't it just run out of memory again, unless they edit this file? Maybe we should allow the  `-M` to be paramterized in this script?

Reply:
they'll rerun the command with a different -M <memory>

Reply:
just want to not specify any memory param by default

Reply:
actually i'm not sure what should be the best default behavior. If only we can find a way to assign the right size... 64GB might be too much and would delay the job if there are no resources in the EBI cluster

Reply:
We have a way to assign size. I have written some code, don't know if I disrupted this script too much so I am not gonna push. Basically, I have rewritten the function "run_job" to:

```
    def run_job(self, archive_job: ArchiveJob):
        self.write_job_json(archive_job)
        job_memory = self.get_job_memory(archive_job)
        job_bsub_command = self.bsub_command(archive_job, job_memory)
        self.run_command(job_bsub_command, stdout_filename=archive_job.job_stdout_path)
```

And added the function `get_job_memory`:
```
    def get_job_memory(self, archive_job: ArchiveJob):
        s3 = boto3.client('s3',
                          aws_access_key_id=self.aws_access_key,
                          aws_secret_access_key=self.aws_key_secret
                          )
        job_urls = [file_input['cloud_url'] for file_input in archive_job.job_dict.get('jobs')[0]['conversion']['inputs']]
        file_sizes = [s3.head_object(Bucket=job_url.split('/')[2], Key="/".join(job_url.split('/')[3:]))['ContentLength']
                      for job_url in job_urls]
        return int(max(file_sizes) / 1000**2 + 10000)
```

Of course, this means importing `boto3` and adding an extra argument to the function `bsub_command`.

I have not commited but I'd like your opinion

Reply:
@ESapenaVentura Nice, I'm happy to include that function for determining memory.